lte/gateway/c/oai/tasks/sgw/pgw_handlers.c

```diff
@@ -536,82 +563,138 @@ uint32_t pgw_handle_nw_initiated_bearer_deactv_req(
     OAILOG_ERROR(
       LOG_PGW_APP, "hashtblP is NULL for nw_initiated_deactv_bearer_req\n");
     OAILOG_FUNC_RETURN(LOG_PGW_APP, RETURNerror);
   }
 
-  //Check if EBI recvd == LBI to know if default bearer has to be deactivated
+  //Check if valid LBI and EBI recvd
   while ((num_elements < hashtblP->num_elements) && (i < hashtblP->size) &&
-         (!found)) {
+         (!is_ebi_found)) {
     pthread_mutex_lock(&hashtblP->lock_nodes[i]);
     if (hashtblP->nodes[i] != NULL) {
       node = hashtblP->nodes[i];
       spgw_ctxt_p = node->data;
       num_elements++;
       if (spgw_ctxt_p != NULL) {
         if (!strcmp(
-              (const char *)
-                spgw_ctxt_p->sgw_eps_bearer_context_information.imsi.digit,
-              (const char *) bearer_req_p->imsi)) {
-          itti_s5_deactv_ded_bearer_req->s11_mme_teid =
+          (const char *)
+          spgw_ctxt_p->sgw_eps_bearer_context_information.imsi.digit,
+          (const char *) bearer_req_p->imsi)) {
+          is_imsi_found = true;
+          s11_mme_teid =
             spgw_ctxt_p->sgw_eps_bearer_context_information.mme_teid_S11;
-          for (j = 0; j < bearer_req_p->no_of_bearers; j++) {
-            if (
-              bearer_req_p->ebi[j] ==
-              spgw_ctxt_p->sgw_eps_bearer_context_information.pdn_connection
-                .default_bearer) {
-              itti_s5_deactv_ded_bearer_req->delete_default_bearer = true;
-              found = true;
-              break;
+          if ((bearer_req_p->lbi != 0) &&
+            (bearer_req_p->lbi ==
+            spgw_ctxt_p->sgw_eps_bearer_context_information.pdn_connection
+              .default_bearer)) {
+            is_lbi_found = true;
+            //Check if the received EBI is valid
+            for (itrn = 0; itrn < bearer_req_p->no_of_bearers; itrn ++) {
+              if(sgw_cm_get_eps_bearer_entry(
+                &spgw_ctxt_p->sgw_eps_bearer_context_information.pdn_connection,
+                bearer_req_p->ebi[itrn])) {
+                is_ebi_found = true;
+                ebi_to_be_deactivated[no_of_bearers_to_be_deact] =
+                  bearer_req_p->ebi[itrn];
+                no_of_bearers_to_be_deact++;
+              } else {
+                invalid_ebi[no_of_bearers_rej] = bearer_req_p->ebi[itrn];
+                no_of_bearers_rej++;
+              }
             }
+          } else {
+            invalid_ebi[no_of_bearers_rej] = bearer_req_p->lbi;
+            no_of_bearers_rej++;
+            OAILOG_ERROR(
+              LOG_PGW_APP,
+              "Unknown LBI (%d) received in pgw_nw_init_deactv_bearer_request"
+              "\n",
+              bearer_req_p->lbi);
           }
+        } else {
+          invalid_ebi[no_of_bearers_rej] = bearer_req_p->lbi;
+          no_of_bearers_rej++;
+          OAILOG_ERROR(
+            LOG_PGW_APP,
+            "Unknown IMSI (%s) received in pgw_nw_init_deactv_bearer_request"
+            "\n",
+            bearer_req_p->imsi);
```

Comment:
I dont see why this is an error. IMSI does not match and we move on to the next IMSI!!